route/donation-router.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const { Router } = require('express');
+const httpErrors = require('http-errors');
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+const Donation = require('../model/donation.js');
+const Profile = require('../model/profile.js');
+
+let fuzzy = (filterTerm) => new RegExp('.*' + filterTerm.toLowerCase().split('').join('.*') + '.*');
```

Comment:
This should be moved to a `lib/fuzzy.js` because its being implamented in more than one file